Irrgarten/src/irrgarten/Game.java

```diff
@@ -0,0 +1,112 @@
+package irrgarten;
+import java.util.ArrayList;
+
+public class Game {
+    public static int MAX_ROUNDS = 10;
+    private int currentPlayerIndex;
+    private String log;
+
+    private Player currentPlayer;
+    private Labyrinth labyrinth;
+    private ArrayList<Monster> monsters;
+    private ArrayList<Player> players;
+
+    // Parametrized
+    public Game(ArrayList<Player> players, ArrayList<Monster> monsters) {
+	this.players = players;
+	this.monsters = monsters;
+	// Escojo aleatoriamente un jugador para empezar
+	this.currentPlayerIndex = Dice.whoStarts(players.size());
+	this.currentPlayer = this.players.get(this.currentPlayerIndex);
+	
+	this.log = "";
+	this.labyrinth = new Labyrinth();
+	this.configureLabyrinth();
+    }
+
+    public boolean finished(){
+	return this.labyrinth.haveAWinner();
+    }
+
+    public boolean nextStep(Directions preferredDirection) {
+        throw new UnsupportedOperationException();
+    }
+
+    public GameState getGameState() {
+	String labyrinth_s = labyrinth.toString();
+	String players_s = players.toString();
+	String monsters_s = monsters.toString();
+	int currentP = this.currentPlayerIndex;
+	boolean winner = this.finished();
+
+	GameState state = new GameState(labyrinth_s, players_s, monsters_s, currentP, winner, this.log);
+
+	return state;
+    }
+
+    private void configureLabyrinth(){
+	// Luego lo escribimos para configurar 
+	// un laberinto concreto
```

Comment:
```suggestion
Monster goblin = new Monster ("Goblin", Dice.randomIntelligence(), Dice.randomStrength());
        labyrinth.addMonster(2,2,goblin);
        monsters.add(goblin);
```

Esto puse yo, es un laberinto simple

Reply:
Está bien, sin más. Un poco arbitrario el nombre del monstruo.